File: dotnet/DesignPatterns/BehavioralDesignPatterns/StrategyDesignPattern/StrategyDesignPattern.Example1/PaymentContext.cs
namespace StrategyDesignPattern.Example1;

public class PaymentContext
{
    IPaymentStrategy _paymentStrategy;

    public PaymentContext(IPaymentStrategy paymentStrategy)
    {
        _paymentStrategy = paymentStrategy;
    }

    public void SetPaymentStrategy(IPaymentStrategy paymentStrategy)
    {
        _paymentStrategy = paymentStrategy;
    }

    public void Pay(decimal amount)
    {
        _paymentStrategy.Pay(amount);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/StrategyDesignPattern/StrategyDesignPattern.Example1/src/PaymentContext.ts
import { IPaymentStrategy } from './IPaymentStrategy';

export class PaymentContext {
  private _paymentStrategy: IPaymentStrategy;

  constructor(paymentStrategy: IPaymentStrategy) {
    this._paymentStrategy = paymentStrategy;
  }

  setPaymentStrategy(paymentStrategy: IPaymentStrategy): void {
    this._paymentStrategy = paymentStrategy;
  }

  pay(amount: number): void {
    this._paymentStrategy.pay(amount);
  }
}
